Realm/Realm.SourceGenerator/Realm.SourceGenerator.csproj

```diff
@@ -0,0 +1,43 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+	<PropertyGroup>
+		<TargetFramework>netstandard2.0</TargetFramework>
+		<AssemblyName>Realm.SourceGenerator</AssemblyName>
+		<PackageId>Realm.SourceGenerator</PackageId>
+		<Title>Realm.SourceGenerator</Title>
+		<ReleaseNotes>https://docs.mongodb.com/realm-sdks/dotnet/latest/CHANGELOG.html</ReleaseNotes>
+		<CodeAnalysisRuleSet>$(ProjectDir)..\..\global.ruleset</CodeAnalysisRuleSet>
+		<DisableFody>true</DisableFody>
+		<NoWarn>1701;1702;NU1701</NoWarn>
+		<LangVersion>10.0</LangVersion>
+		<IsRoslynComponent>true</IsRoslynComponent>
+	</PropertyGroup>
+	<ItemGroup>
+		<!-- Unity does not support the latest version of CodeAnalysis.CSharp. According to the docs it should support version 3.8.0, 
+		but it works also with 3.9.0. We need to verify manually if it works also with some later versions. -->
+		<PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.9.0" PrivateAssets="all" />
+		<PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.3">
+			<PrivateAssets>all</PrivateAssets>
+			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+		</PackageReference>
```

Comment:
Can we add `Microsoft.CodeAnalysis.FxCopAnalyzers` and `StyleCop.Analyzers` so that we get warnings about style violations?